RFC for ecommerce REST APIs

noisecapella opened the pull request:
#### What are the relevant tickets?
Part of #75 

#### What's this PR do?
This describes some of the REST APIs which will be implemented for ecommerce.

mbertrand commented on docs/rfcs/0005-ecommerce-rest-api.md:
```diff
@@ -0,0 +1,182 @@
+
+    {
+        "coupons": []
+    }
```
If a user manually applies a coupon code, then clears that coupon, should the best available automatic coupon for the basket product be applied again?

noisecapella replied:
Yes I think that's right. I'll add some text to make that explicit.

mbertrand commented on docs/rfcs/0005-ecommerce-rest-api.md:
```diff
@@ -0,0 +1,182 @@
+                "code": "DISCOUNT_30_OFF"
+            }
+        ]
+    } 
```
Seems like the `api/basket` API will handle all coupon-related functions including validation, which makes sense, so there's probably no need for #71 as a separate issue anymore?

noisecapella replied:
Hmm I think you're right

mbertrand replied:
If you'd like to split up the implementation of this differently (`api/basket` vs `api/products` vs `api/checkout` etc), let me know what part(s) you'd prefer me to work on.

mbertrand commented on docs/rfcs/0005-ecommerce-rest-api.md:
```diff
@@ -0,0 +1,182 @@
+A user could update the basket to have one item by patching:
+
+    {
+        "items": [
```
Will a patch request with multiple items be invalidated?

noisecapella replied:
Yes, we should probably make that an error

noisecapella commented:
I addressed the review feedback, and I also added `type` and `course_runs` to the product schema to provide more information when showing the basket at checkout.

pdpinch commented on docs/rfcs/0005-ecommerce-rest-api.md:
```diff
@@ -0,0 +1,208 @@
+### /api/checkout/
+
+Unlike MicroMasters there is no payload required for the checkout API. Instead it will
+use the user's basket to figure out what should be purchased.
```
I want to be sure that we can still pass order data to the merchant-defined fields in cybersource.

noisecapella replied:
From the docs it looks like we can have up to 100 merchant-defined fields so the number of items in the basket will need to be limited depending on how many we use per basket item.

noisecapella replied:
Actually it looks like the first four are stored differently from the other 96, though I'm not sure what the significance is for us:

>Merchant-defined data fields 1 to 4 are stored against the payment token and are used for subsequent token based transactions. Merchant defined data fields 5 to 100 are passed trough to Decision Manager as part of the initial payment request and are not stored against the payment token. 
From page 96 of [the docs](http://apps.cybersource.com/library/documentation/dev_guides/Secure_Acceptance_Hosted_Checkout/Secure_Acceptance_Hosted_Checkout.pdf)
